src/lib/reader.c

```diff
@@ -203,6 +203,7 @@ pgp_reader_push(pgp_stream_t *          stream,
 
     if ((readinfo = calloc(1, sizeof(*readinfo))) == NULL) {
         (void) fprintf(stderr, "pgp_reader_push: bad alloc\n");
+        return 0;
```

Comment:
I think where places we just `return` in an if-else block should just be short circuited like this?

```c
if (x) {
  ...
  return 0;
} else {
  ...
}
```

To:
```c
if (x) {
  ...
  return 0;
}
...
```